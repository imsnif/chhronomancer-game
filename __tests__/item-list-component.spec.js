import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

const items = [
  { name: 'assist', source: false, count: 1 },
  { name: 'prevent', source: 'Timeline 3', count: 2 },
  { name: 'reset', source: 'Timeline 4', count: 2 },
  { name: 'steal', source: 'Timeline 1', count: 1 },
  { name: 'lock', source: false },
  { name: 'unlock', source: false, count: 0 }
]

test('Item list => renders correctly', () => {
  const ItemList = require('../components/item-list').default
  const tree = renderer.create(
    <ItemList itemCount={6} items={items} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
