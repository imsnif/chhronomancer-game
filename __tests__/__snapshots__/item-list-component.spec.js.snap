// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item list => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "paddingLeft": 5,
          "textAlign": "left",
        }
      }
    >
      Items
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 3,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      Cap: 6/8
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#00ff00",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Steal
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Assist
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#00ff00",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Prevent (x2)
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#00ff00",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Reset (x2)
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Lock
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "height": 48.690000000000005,
          "marginTop": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 0,
            "height": 46.690000000000005,
            "marginLeft": 4,
            "resizeMode": "stretch",
            "width": 46.690000000000005,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 5,
              "fontFamily": "telegrama_raw",
              "paddingLeft": 9,
              "paddingTop": 3,
            }
          }
        >
          Unlock
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#329932",
              "flex": 1,
              "fontFamily": "telegrama_raw",
              "paddingTop": 5,
            }
          }
        >
          ?
        </Text>
      </View>
    </View>
  </View>
</View>
`;
